Proj conf

yarondbb opened the pull request:
Params new capability to load from yml
Logger corrections (sorry mistakenly I have worked on this on the same branch..)

yarondbb committed:
refactor tests

yarondbb committed:
Added capability to load and overwrite params via yml file
Added test to params to check the new capability

yarondbb committed:
Changed yml foramt to be a simple hash of param=value
Added check that If yml param does not exist in Params Module then the method will fail
Added tests for the above behavior

kolmanv commented on lib/params.rb:
```diff
@@ -7,7 +7,20 @@ def parameter(name, default_value, description)
+    def read_from_yml file_path
+      proj_params = YAML::load_file(file_path)
+      proj_params.keys.each do
+      |key|
```
move |key| to previous line.
.each do |key|

kolmanv commented on test/log/log_test.rb:
```diff
@@ -14,24 +14,23 @@ module Log
     end
 
     class TestLog < Test::Unit::TestCase
+
```
Remove this line.

kolmanv commented on test/params/test_1.yml.bak:
```diff
@@ -0,0 +1,7 @@
```
Remove this file from your commit.

kolmanv commented on test/params/params_test.rb:
```diff
@@ -0,0 +1,42 @@
+      code_param_name =  'param_1'
+      code_param_value = 'value_1_code'
+      yml_param_value = 'value_1_yml'
+      yml_file_name = 'test/params/test_1.yml'
```
You don't have to work with files here.
You can inline write content and load yml from string.
This is better because the test is more dynamic, you don't have to edit few files if things change.
Better remove the yml test files (unless you really have a good reason to leave them) and test with strings.
This is always the preference.
Just do Yaml.load(yaml string)

kolmanv commented:
Yaron, please notice that you had to create two branches here.
I want to merge the log fixes you did well, but I cannot because there are not ready things in Params.
GIT enables working with many many branches in parallel. This is good because of concern separation.

kolmanv commented:
Ohhh, now I see your comment at the beginning. Disregard the previous comment.

yarondbb committed:
Removed test files. Using yml string instead
read_yml_params will return true if all parameters loaded successfully. False otherwise.

kolmanv commented:
Yaron, did you fix this pull request? May I merge?

yarondbb commented:
Yes, I fixed the issues. you can merge
1 question: if I do not test load from file then how would I add a method to load from file?
